File: peer1_logan/advertise.rb
require "zeroconf"

# Advertise the peer over mDNS as _peer._tcp.local.
Zeroconf.service "_peer._tcp.local.", 3002, "peer1-ruby"

# Keep the script running so the service stays discoverable
puts "📡 Advertising peer1-ruby on _peer._tcp.local.:3002"
sleep

File: discovery/advertise.js
const bonjour = require('bonjour')();

const service = bonjour.publish({
  name: 'peer1-ruby',
  type: 'p2p',
  port: 3003
});

console.log('📡 Advertising peer1-ruby._p2p._tcp.local on port 3003');
